### PR TITLE
Fix behavior of hostfs under agent

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This is a fix for : https://github.com/elastic/beats/issues/28459

Under agent, `system.hostfs` doesn't seem to get applied to individual beats. The config is passed along, but never instantiated. I've given up trying to figure out how config gets from agent to metricbeat, but this is at least a temporary fix that re-instantiates the `hostfs` value in the main system module code. We need something similar in the linux module, but that code needs enough change that I want to do that in a different PR. I'm not sure when this bug was introduced or what changed. There might be a more elegant way to fix this, but I don't know enough about how agent "reloads" beat configs to figure out what that is, so this is a bit of an awkward fix.

## Why is it important?

`system.hostfs` is broken under agent, which is bad.

## Checklist
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

This is honestly kind of a pain to test, particularly with `elastic-package`. 
- Build agent along with metricbeat
- run and enroll agent, add `/hostfs` or any junk value to the system module agent config for the system root.
- Check the logs, If the agent suddenly breaks, that means it works.

### CODE REVIEW

**Comment by ruflin** [comment]
@michalpristas Can you help out @fearful-symmetry on how the configs get from agent to metricbeat? It does not mean we should blockt his PR on it but I think it is still worth sharing.

**Review by faec — APPROVED** [review]
(after a long offline discussion of the issue and context) This PR makes me uncomfortable, but it's a reasonable short-term fix given how we've painted ourselves in a corner. Approved, but I hope we can make it a priority to rethink `hostfs` behavior and config in early 8.x.

**Comment by jlind23** [comment]
@faec @fearful-symmetry do we already have an issue to fix that "technical debt" later on? If no, i'll be happy to create it and will add it then to our project board.

**Comment by ruflin** [comment]
@jlind23 Yes: https://github.com/elastic/beats/issues/28706 This recently came up also in other conversations around global variables.

**Comment by DaveSys911** [comment]
> @faec @fearful-symmetry do we already have an issue to fix that "technical debt" later on? If no, i'll be happy to create it and will add it then to our project board.

Could you please link it here ? i'd like to follow that discussion

**Comment by fearful-symmetry** [comment]
@faec so, because this is so ugly, I made a few changes to clean up some unnecessary `hostfs` references.  At least, this will make the code a tad more consistent and limit the cases outside of the system module here we need to access the global variable.

File: metricbeat/module/system/system.go [hunk 1]
```diff
@@ -51,7 +51,9 @@ type SystemModule interface {
 
 func NewModule(base mb.BaseModule) (mb.Module, error) {
 	var hostfs string
+
 	// If this is fleet, ignore the global path, as its not being set.
+	// This is a temporary hack
```
Comment (faec) [inline metricbeat/module/system/system.go:56]: [you hear the distant sound of gods laughing]

**Review by faec — APPROVED** [review]
Yea this will make things slightly more manageable later, looks good

**Pull request merged**